### PR TITLE
Features

### PR DESCRIPTION
**Pull request by mpeddycord**
## Summary
Added "Bro Split" (Bodybuilding body-part split (Chest/Back/Shoulders/Arms/Legs)) as option during onboarding
Updates to rest times to more appropriately apply rest periods to blocks

## Changes

- Added "Bro Split" (Bodybuilding body-part split (Chest/Back/Shoulders/Arms/Legs)) as option during onboarding
- Added bodybuilding prompt section that helps the AI generate appropriate plan for new plan type
- Rep Schemes / Session Types focusing on specific body parts each day as opposed to just upper/lower/full body
- BUG FIX: mapped REP_SET_SCHEMES to goal scheme. Previously number of sets was hard-coded to "3" 
- Updated .env.example for JWT_PRIVATE_KEY/JWKS/SITE_URL to allow quicker usage by other developers

## Checklist

- [X] `npx tsc --noEmit` passes
- [X] `npm run lint` passes
- [X] `npm test` passes (existing + new tests)
- [X] New logic has tests (happy path + at least one error/edge case)
- [X] No new or materially expanded file exceeds the 300-line soft cap
- [X] No file exceeds the 400-line hard cap (enforced by CI)
- [X] Functions stay near the 60-line convention
- [X] No new `any` types (enforced by ESLint); `as` casts only at deserialization boundaries
- [X] Tested in browser (if UI changes)
- [X] Commits follow conventional format (`type: description`)
- [X] No unused exports or dead code introduced

## Test Plan

- Added additional tests to goalConfig Tests.
- Tested via UI extensively

## Summary by CodeRabbit

* **New Features**
  * Added "Bro Split" (Chest / Back / Shoulders / Arms / Legs) as a training program option
  * Added "Bodybuilding / aesthetics" as a goal selection
  * Support for 6-day training weeks
  * Bodybuilding mode with specialized coaching guidance, including volume targets (15–20 sets/muscle/week for priority groups) and isolation emphasis

* **Improvements**
  * Enhanced authentication flow with improved redirect logic
  * Better muscle readiness display labels with dynamic verb conjugation

### CODE REVIEW

File: convex/coach/weekProgrammingHelpers.ts [hunk 1]
```diff
@@ -151,6 +146,16 @@ export function getSessionTypesForSplit(
...
   }
+  if (split === "bro_split") {
+    // Classic bodybuilding body-part split: chest → back → shoulders → arms → legs.
+    // Capped at 5 days — there are only 5 distinct body parts, so days beyond that
+    // become rest days rather than cycling back to chest.
+    const types: SessionType[] = ["chest", "back", "shoulders", "arms", "legs"];
+    return trainingDayIndices.slice(0, types.length).map((dayIndex, i) => ({
+      dayIndex,
+      sessionType: types[i],
+    }));
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/coach/weekProgrammingHelpers.ts:157]: Not a bug -- this is intentional, commented, and tested. The test `clamps bro_split to 5 days when >5 training days are requested` validates the behavior. The caller fills extra days with `"rest"` via `sessionTypeByDay.get(i) ?? "rest"`. A bro split has exactly 5 body parts; cycling back to chest would be incorrect programming.

File: convex/coach/workoutBlocks.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    }
+    groupedByAccessory.get(accessory)!.push(movementId);
+  }
+
+  const buildExercise = (movementId: string) => {
+    const movement = catalogMap.get(movementId);
+    const isDurationBased = movement ? !movement.countReps : false;
+    if (isDurationBased) {
+      return { movementId, sets: baseSets, duration: DEFAULT_DURATION_SECONDS };
+    }
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/coach/workoutBlocks.ts:84]: Fixed. `buildExercise` now uses `options?.goalScheme?.duration ?? DEFAULT_DURATION_SECONDS`, so duration-based goals like `mobility_flexibility` (35s) are respected. Added tests in `blocksFromMovementIds.test.ts` covering goalScheme.duration present, absent, and omitted.

File: src/app/login/page.tsx [hunk 3]
```diff
@@ -36,17 +36,22 @@ export default function LoginPage() {
...
   const [submitting, setSubmitting] = useState(false);
+  const authenticatedRedirectPath = flow === "signUp" ? "/onboarding" : "/chat";
 
   useEffect(() => {
     if (!authLoading && isAuthenticated) {
-      router.replace("/chat");
+      router.replace(authenticatedRedirectPath);
     }
-  }, [authLoading, isAuthenticated, router]);
+  }, [authLoading, isAuthenticated,authenticatedRedirectPath, router]);
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply src/app/login/page.tsx]: Fixed. Redirect path is now captured in a `useRef` at submit time, so toggling the sign-in/sign-up tab mid-auth cannot change the destination. The effect falls back to `/chat` when the ref is unset (matching the previous default behavior for already-authenticated users).

File: src/features/chat/ToolCallIndicator.tsx [hunk 4]
```diff
@@ -123,7 +123,7 @@ export function ToolCallIndicator({ toolName, state, output }: ToolCallIndicator
     if (data.success && data.summary) {
       const plan: WeekPlanPresentation = {
         weekStartDate: data.summary.weekStartDate,
-        split: data.summary.preferredSplit as "ppl" | "upper_lower" | "full_body",
+        split: data.summary.preferredSplit as "ppl" | "upper_lower" | "full_body" | "bro_split",
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply src/features/chat/ToolCallIndicator.tsx]: Fixed. Removed the unsafe `as` cast. The `split` value is now validated via `weekPlanPresentationSchema.shape.split.safeParse()`, and the rest of the output is defensively accessed as `Record<string, unknown>`. Invalid splits return `null` instead of rendering garbage. Added a test for the invalid-split case.

File: src/features/dashboard/MuscleReadinessMap.tsx [hunk 5]
```diff
@@ -67,29 +71,30 @@ export function MuscleReadinessMap({ readiness }: MuscleReadinessMapProps) {
           const fresh = entries.find((e) => e.value > 80);
           if (!fresh) return null;
           const prompt = encodeURIComponent(
-            `My ${fresh.muscle.toLowerCase()} is at ${fresh.value}% readiness. Can you program a ${fresh.muscle.toLowerCase()} workout?`,
+            `My ${fresh.muscle.toLowerCase()}&nbsp;${isOrAre(fresh.muscle)}&nbsp; at ${fresh.value}% readiness. Can you program a ${fresh.muscle.toLowerCase()}&nbsp; workout?`,
           );
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply src/features/dashboard/MuscleReadinessMap.tsx]: Fixed. Replaced literal `&nbsp;` in template strings with regular spaces. Added a test verifying `decodeURIComponent(href)` does not contain `&nbsp;`.

**Review by JeffOtano — APPROVED** [review]

**Comment by JeffOtano** [comment]
Thanks @mpeddycord, you're the first external contributor! 
Really appreciate you jumping in. Excited to see what you build next.
